crates/swc_css_minifier/src/compressor/calc_sum.rs

```diff
@@ -0,0 +1,749 @@
+                    CalcValueOrOperator::Value(calc_value) => Some(calc_value.clone()),
+                    _ => None,
+                }
+            }
+            _ => None,
+        };
+    }
+
+    None
+}
+
+fn transform_calc_value_into_component_value(calc_value: &CalcValue) -> ComponentValue {
+    match &calc_value {
+        CalcValue::Number(n) => ComponentValue::Number(n.clone()),
+        CalcValue::Dimension(Dimension::Length(l)) => {
+            ComponentValue::Dimension(Dimension::Length(Length {
+                span: l.span,
+                value: l.value.clone(),
+                unit: l.unit.clone(),
+            }))
+        }
+        CalcValue::Dimension(Dimension::Angle(a)) => {
+            ComponentValue::Dimension(Dimension::Angle(Angle {
+                span: a.span,
+                value: a.value.clone(),
+                unit: a.unit.clone(),
+            }))
+        }
+        CalcValue::Dimension(Dimension::Time(t)) => {
+            ComponentValue::Dimension(Dimension::Time(Time {
+                span: t.span,
+                value: t.value.clone(),
+                unit: t.unit.clone(),
+            }))
+        }
+        CalcValue::Dimension(Dimension::Frequency(f)) => {
+            ComponentValue::Dimension(Dimension::Frequency(Frequency {
+                span: f.span,
+                value: f.value.clone(),
+                unit: f.unit.clone(),
+            }))
+        }
+        CalcValue::Dimension(Dimension::Resolution(r)) => {
+            ComponentValue::Dimension(Dimension::Resolution(Resolution {
+                span: r.span,
+                value: r.value.clone(),
+                unit: r.unit.clone(),
+            }))
+        }
+        CalcValue::Dimension(Dimension::Flex(f)) => {
+            ComponentValue::Dimension(Dimension::Flex(Flex {
+                span: f.span,
+                value: f.value.clone(),
+                unit: f.unit.clone(),
+            }))
+        }
+        CalcValue::Dimension(Dimension::UnknownDimension(u)) => {
+            ComponentValue::Dimension(Dimension::UnknownDimension(UnknownDimension {
+                span: u.span,
+                value: u.value.clone(),
+                unit: u.unit.clone(),
+            }))
+        }
+        CalcValue::Percentage(p) => ComponentValue::Percentage(Percentage {
+            span: p.span,
+            value: p.value.clone(),
+        }),
+        CalcValue::Constant(c) => ComponentValue::Ident(c.clone()),
+        CalcValue::Function(f) => ComponentValue::Function(Function {
+            span: f.span,
+            name: f.name.clone(),
+            value: f.value.to_vec(),
+        }),
+        CalcValue::Sum(_) => {
+            unreachable!("CalcValue::Sum cannot be transformed into a ComponentValue")
+        }
+    }
+}
+
+fn fold_calc_sum(calc_sum: &mut CalcSum) {
+    // TODO we should sort the operands by unit to handle this:
+    // "4px + 3% + 5px + 10%" = "9px + 13%"
```

Comment:
Do we have tests case for this for future improvements?

Reply:
Yes we hate this test: `width:calc((100px - 1em) + (-50px + 1em))`